README.md

```diff
@@ -33,108 +34,242 @@ Menu
 
-Save it as `script.js`, then run k6 like this:
+For a complete list of supported k6 options, take a look at the documentation at [https://docs.k6.io/docs/options](https://docs.k6.io/docs/options).
 
-`k6 run script.js`
+_Hint: besides accessing the supplied [environment variables](https://docs.k6.io/docs/environment-variables) through the `__ENV` global object briefly mentioned above, you can also use the [execution context variables](https://docs.k6.io/docs/execution-context-variables) `__VU` and `__ITER` to access the current VU number and the number  oof the current iteration **in that VU**. These variables can be very useful if you want to have different scenarios for different VUs or you want to generate different data from different VUs: ```http.post("https://some.example.website/signup", {username: `testuser${__VU}@testsite.com`, /* ... */})```_
 
-(Note that if you use the Docker image, the command is slightly different: `docker run -i loadimpact/k6 run - <script.js`)
+For even more complex scenarios, you can use the k6 [REST API](https://docs.k6.io/docs/rest-api) and the `k6 status`, `k6 scale`, `k6 pause`, `k6 resume` CLI commands to manually control a running k6 test. For [cloud-based tests](https://docs.k6.io/docs/cloud-execution), executed on Load Impact's managed infrastructure via the `k6 cloud` command, you can also specify the VU distribution percentages for different load zones when executing load tests, giving you scalable and geographically-distributed test execution.
 
-For more information on how to get started running k6, please look at the [Running k6](https://docs.k6.io/docs/running-k6) documentation. If you want more info on the scripting API or results output, you'll find that also on [https://docs.k6.io](https://docs.k6.io).
 
----
+### Setup and teardown
+
+Beyond the init code and the required VU stage (i.e. the `default` function), which is code run for each VU, k6 also supports test wide setup and teardown stages, like many other testing frameworks and tools. The `setup` and `teardown` functions, like the `default` function, need to be `export`ed. But unlike the `default` function, `setup` and `teardown` are only called once for a test - `setup()` is called at the beginning of the test, after the init stage but before the VU stage (`default` function), and `teardown()` is called at the end of a test, after the last VU iteration (`default` function) has finished executing. This works even when executing a test in the distributed [cloud execution](https://docs.k6.io/docs/cloud-execution) mode via `k6 cloud`.
+
+See the [full example below](#example-code) for an illustration of how `setup()` and `teardown()` work, how they can be useful in a realistic situation, and how you can pass information from `setup()` to the `default` function and `teardown()`. You can also find more information and examples in the k6 docs [here](https://docs.k6.io/docs/test-life-cycle#section-setup-and-teardown-stages).
```

Comment:
See the [full example below](#example-code) for an illustration of how `setup()` and `teardown()` work, how they can be useful in a realistic situation, and how you can pass information from `setup()` to the `default` function and `teardown()`. For more information and examples, refer to the k6 docs [here](https://docs.k6.io/docs/test-life-cycle#section-setup-and-teardown-stages).